### ISSUE TITLE
Prevents parallel Babel (still? again? not sure ¯\_(ツ)_/¯)

### ISSUE THREAD

**Issue by dfreeman** [ember-template-lint/ember-cli-template-lint#512]
#444 was originally opened for this and #480 was merged to address it, but it seems as though the expectations for parallelization config for template transforms are different in Embroider vs vanilla ember-cli.

I _believe_ the return value from `buildUsing` is expected to be an object in the format you'd pass to `registry.add`, not just the AST plugin, i.e. something roughly like:

```js
  setupPreprocessorRegistry(type, registry) {
    registry.add('htmlbars-ast-plugin', this.buildPlugin());
  },

  buildPlugin() {
    let RemoveConfigurationHtmlComments = require('./lib/plugins/remove-configuration-html-comments');
    return {
      name: 'remove-configuration-html-commeents',
      plugin: RemoveConfigurationHtmlComments(),
      baseDir: () => __dirname,
      parallelBabel: {
        requireFile: __filename,
        buildUsing: 'buildPlugin'
      }
    };
  }
```

**Comment by jmonster** [comment]
This issue is alive and kickin' when you generate a brand new ember-cli app. I isolated the issue down to this plugin before finding this issue 🤦‍♂ 



## with ember-cli-template-lint
```sh
~/D/wubalubadubdub ❯❯❯ yarn add ember-cli-template-lint
yarn add v1.21.1
[1/5] 🔍  Validating package.json...
.....
```
```sh
~/D/wubalubadubdub ❯❯❯ env THROW_UNLESS_PARALLELIZABLE=1 yarn start
yarn run v1.21.1
$ ember serve
[broccoli-persistent-filter:Babel > [Babel: wubalubadubdub]: Babel: wubalubadubdub] was configured to `throwUnlessParallelizable` and was unable to parallelize a plugin.
plugins:
1: name: unknown, location: unknown

Please see: https://github.com/babel/broccoli-babel-transpiler#parallel-transpilation for more details


Stack Trace and Error Report: .....
```

## without ember-cli-template-lint
```sh
~/D/wubalubadubdub ❯❯❯ yarn remove ember-cli-template-lint
yarn remove v1.21.1
[1/2] 🗑  Removing module ember-cli-template-lint...
.....
```
```sh
~/D/wubalubadubdub ❯❯❯ env THROW_UNLESS_PARALLELIZABLE=1 yarn start
yarn run v1.21.1
$ ember serve

Build successful (2271ms) – Serving on http://localhost:4200/
.....
```

**Comment by patocallaghan** [comment]
@jmonster I've created a PR which fixes this https://github.com/ember-template-lint/ember-cli-template-lint/pull/579
